pin distopia<0.3.0

orbeckst opened the pull request:
Fixes #4739

Changes made in this Pull Request:
- [x] temporarily restrict distopia in CI to >=0.2.0,<0.3.0 until MDAnalysis has caught up with distopia API changes
- [x] updated CHANGELOG
- [x] version check in code

PR Checklist
------------
 - [x] Tests?
 - [ ] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

## Developers certificate of origin
- [x] I certify that this contribution is covered by the LGPLv2.1+ license as defined in our [LICENSE](https://github.com/MDAnalysis/mdanalysis/blob/develop/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).


<!-- readthedocs-preview mdanalysis start -->
----
📚 Documentation preview 📚: https://mdanalysis--4740.org.readthedocs.build/en/4740/

<!-- readthedocs-preview mdanalysis end -->

orbeckst committed:
pin distopia<0.3.0

- fix  #4739
- temporarily restrict distopia to >=0.2.0,<0.3.0 until MDAnalysis has
  caught up with distopia API changes
- updated CHANGELOG

orbeckst commented on package/CHANGELOG:
```diff
@@ -78,6 +78,7 @@ Enhancements
    DOI 10.1021/acs.jpcb.7b11988. (Issue #2039, PR #4524)
 
 Changes
+ * only use distopia < 0.3.0 due to API changes (Issue #4739)
```
Not sure if we would normally make a CHANGELOG entry but in this case I would recommend to do so because there is not other part in the user-visible code base that knows about this requirement. If someone installs distopia by themselves, they will see failures.

orbeckst replied:
Now they will (hopefully) a warning
```
../package/MDAnalysis/lib/_distopia.py:45
  ~/MDAnalysis/mdanalysis/package/MDAnalysis/lib/_distopia.py:45: RuntimeWarning: Install 'distopia>=0.2.0,<0.3.0' to be used with this release of MDAnalysis. Your installed version of distopia >=0.3.0 will NOT be used.
    warnings.warn("Install 'distopia>=0.2.0,<0.3.0' to be used with this release "
```

orbeckst commented on .github/actions/setup-deps/action.yaml:
```diff
@@ -1,4 +1,4 @@
-name: 'setup-deps'
+ame: 'setup-deps'
```
oops ... will fix

hmacdope reviewed (approved):
LGTM @orbeckst, thanks for fixing.

hmacdope commented:
Merge when ready

orbeckst commented:
Let's hold on merging. I think @IAlibay also wanted a version check in the code and I agree as this will at least catch the issue with users installing distopia >= 0.3.0. Given the loose coupling between MDA and distopia, it seems necessary and you can then recycle it immediately for PR #4734 .

orbeckst committed:
version check distopia

- only enable HAS_DISTOPIA if the appropriate version of distopia is installed
- issue RuntimeWarning if incorrect version present
- added test

pep8speaks commented:
Hello @orbeckst! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:





There are currently no PEP 8 issues detected in this Pull Request. Cheers! :beers: 

##### Comment last updated at 2024-10-19 01:11:18 UTC

hmacdope commented:
I will make a patch release for `distopia` that has `__version__`, sorry thats my bad.

orbeckst commented:
I tested locally with distopia==0.2.0 and distopia=0.3.0 and the code behaves as it should: it fails for neither but warns for 0.3.0 and falls back to `HAS_DISTOPIA=False`.

orbeckst commented on package/MDAnalysis/lib/_distopia.py:
```diff
@@ -36,10 +37,22 @@
+    # check for compatibility: currently needs to be >=0.2.0,<0.3.0 (issue
+    # #4740) No distopia.__version__ available so we have to do some probing.
+    needed_funcs = ['calc_bonds_no_box_float', 'calc_bonds_ortho_float']
+    has_distopia_020 = all([hasattr(distopia, func) for func in needed_funcs])
```
There's no `distopia.__version__` attribute that I could check so I am explicitly probing for the functions we need.

@hmacdope is there a better way to get the version of the installed distopia package?

hmacdope replied:
^ See comment above, I will make a patch release, I accidentally removed `__version__`

orbeckst commented on testsuite/MDAnalysisTests/lib/test_distances.py:
```diff
@@ -788,9 +790,10 @@ def test_pbc_wrong_wassenaar_distance(self, backend):
 
-@pytest.mark.parametrize("box", 
+
+@pytest.mark.parametrize("box",
```
pep8 fix

IAlibay reviewed (approved):
Thanks @orbeckst !

orbeckst commented:
I'll merge with the clunky version probe to move things along and if needed we improve when there's a distopia 0.3.1.
